When gtk 3 tries to organize buttons in tabular representation,
like in shutdown/logout dialogs, they look extremely ugly.
Avoid aligning buttons to some particular border makes them look better.
We use 3px border widths as with 7px some artifacts appear.

--- nimbus-gtk3-theme-4.2/nimbus/gtk-3.0/gtk-widgets.css	2016-11-28 17:13:37.136832970 +0300
+++ /export/home/alp/tmp/no-align-gtk-widgets.css	2016-11-28 17:36:16.369958636 +0300
@@ -439,7 +439,7 @@
 .toolbar .linked .button:focus,
 .toolbar .linked .button:focus:active,
 .toolbar .linked .button:insensitive {
-    border-image-width: 7px 0 7px 0;
+    border-image-width: 3px 3px 3px 3px;
 }
 
 /* leftmost button */
@@ -457,7 +457,7 @@
 .toolbar .linked .button:focus:first-child,
 .toolbar .linked .button:focus:active:first-child,
 .toolbar .linked .button:insensitive:first-child {
-    border-image-width: 7px 0 7px 7px;
+    border-image-width: 3px 3px 3px 3px;
 }
 
 /* rightmost button */
@@ -475,7 +475,7 @@
 .toolbar .linked .button:focus:last-child,
 .toolbar .linked .button:focus:active:last-child,
 .toolbar .linked .button:insensitive:last-child {
-    border-image-width: 7px 7px 7px 0;
+    border-image-width: 3px 3px 3px 3px;
 }
 
 /* linked single button */
@@ -493,7 +493,7 @@
 .toolbar .linked .button:focus:only-child,
 .toolbar .linked .button:focus:active:only-child,
 .toolbar .linked .button:insensitive:only-child {
-    border-image-width: 7px 7px 7px 7px;
+    border-image-width: 3px 3px 3px 3px;
 }
 
 /* middle button (vertical) */
@@ -511,7 +511,7 @@
 .toolbar .linked.vertical .button:focus,
 .toolbar .linked.vertical .button:focus:active,
 .toolbar .linked.vertical .button:insensitive {
-    border-image-width: 7px 0 7px 0;
+    border-image-width: 3px 3px 3px 3px;
 }
 
 /* leftmost button (vertical) */
@@ -529,7 +529,7 @@
 .toolbar .linked.vertical .button:focus:first-child,
 .toolbar .linked.vertical .button:focus:active:first-child,
 .toolbar .linked.vertical .button:insensitive:first-child {
-    border-image-width: 7px 7px 0 7px;
+    border-image-width: 3px 3px 3px 3px;
 }
 
 /* rightmost button (vertical) */
@@ -547,7 +547,7 @@
 .toolbar .linked.vertical .button:focus:last-child,
 .toolbar .linked.vertical .button:focus:active:last-child,
 .toolbar .linked.vertical .button:insensitive:last-child {
-    border-image-width: 0 7px 7px 7px;
+    border-image-width: 3px 3px 3px 3px;
 }
 
 /* linked single button (vertical) */
@@ -565,7 +565,7 @@
 .toolbar .linked.vertical .button:focus:only-child,
 .toolbar .linked.vertical .button:focus:active:only-child,
 .toolbar .linked.vertical .button:insensitive:only-child {
-    border-image-width: 7px 7px 7px 7px;
+    border-image-width: 3px 3px 3px 3px;
 }
 
 /* middle button */
@@ -589,7 +589,7 @@
 .inline-toolbar.toolbar GtkToolButton .button:focus,
 .inline-toolbar.toolbar GtkToolButton .button:focus:active,
 .inline-toolbar.toolbar GtkToolButton .button:insensitive {
-    border-image-width: 7px 0 7px 0;
+    border-image-width: 3px 3px 3px 3px;
 }
 
 /* leftmost button */
@@ -613,7 +613,7 @@
 .inline-toolbar.toolbar GtkToolButton:first-child .button:focus,
 .inline-toolbar.toolbar GtkToolButton:first-child .button:focus:active,
 .inline-toolbar.toolbar GtkToolButton:first-child .button:insensitive {
-    border-image-width: 7px 0 7px 7px;
+    border-image-width: 3px 3px 3px 3px;
 }
 
 /* rightmost button */
@@ -637,7 +637,7 @@
 .inline-toolbar.toolbar GtkToolButton:last-child .button:focus,
 .inline-toolbar.toolbar GtkToolButton:last-child .button:focus:active,
 .inline-toolbar.toolbar GtkToolButton:last-child .button:insensitive {
-    border-image-width: 7px 7px 7px 0;
+    border-image-width: 3px 3px 3px 3px;
 }
 
 /* single button */
@@ -661,7 +661,7 @@
 .inline-toolbar.toolbar GtkToolButton:only-child .button:focus,
 .inline-toolbar.toolbar GtkToolButton:only-child .button:focus:active,
 .inline-toolbar.toolbar GtkToolButton:only-child .button:insensitive {
-    border-image-width: 7px 7px 7px 7px;
+    border-image-width: 3px 3px 3px 3px;
 }
 
 
